Compiling with `PETSc` with no MPI

When compiling an `AMReX`-linked code (eg., WarpX) with `PETSc` with GCC (no MPI), I get the following error when `AMReX_PETSc.cpp` is being compiled:
```
In file included from /home/runner/work/warpx/warpx/petsc/include/petscsys.h:64,
                 from /home/runner/work/warpx/warpx/petsc/include/petscvec.h:8,
                 from /home/runner/work/warpx/warpx/petsc/include/petscmat.h:6,
                 from /home/runner/work/warpx/warpx/petsc/include/petscpc.h:6,
                 from /home/runner/work/warpx/warpx/petsc/include/petscksp.h:6,
                 from /home/runner/work/warpx/warpx/build/_deps/fetchedamrex-src/Src/Extern/PETSc/AMReX_PETSc.cpp:9:
/home/runner/work/warpx/warpx/petsc/include/petsc/mpiuni/mpi.h:200:3: error: conflicting declaration ‘typedef struct MPI_Status MPI_Status’
  200 | } MPI_Status;
      |   ^~~~~~~~~~
In file included from /home/runner/work/warpx/warpx/build/_deps/fetchedamrex-src/Src/Base/AMReX.H:9,
                 from /home/runner/work/warpx/warpx/build/_deps/fetchedamrex-src/Src/Base/AMReX_BLassert.H:6,
                 from /home/runner/work/warpx/warpx/build/_deps/fetchedamrex-src/Src/Base/AMReX_Algorithm.H:8,
                 from /home/runner/work/warpx/warpx/build/_deps/fetchedamrex-src/Src/Base/AMReX_BaseFab.H:5,
                 from /home/runner/work/warpx/warpx/build/_deps/fetchedamrex-src/Src/Base/AMReX_FArrayBox.H:6,
                 from /home/runner/work/warpx/warpx/build/_deps/fetchedamrex-src/Src/Extern/HYPRE/AMReX_Habec_K.H:6,
                 from /home/runner/work/warpx/warpx/build/_deps/fetchedamrex-src/Src/Extern/PETSc/AMReX_PETSc.cpp:7:
/home/runner/work/warpx/warpx/build/_deps/fetchedamrex-src/Src/Base/AMReX_ccse-mpi.H:51:8: note: previous declaration as ‘struct MPI_Status’
   51 | struct MPI_Status {};
      |        ^~~~~~~~~~
gmake[2]: *** [_deps/fetchedamrex-build/Src/CMakeFiles/amrex_3d.dir/build.make:1692: _deps/fetchedamrex-build/Src/CMakeFiles/amrex_3d.dir/Extern/PETSc/AMReX_PETSc.cpp.o] Error 1
```

I get this error now:
```
/home/runner/work/warpx/warpx/Source/Initialization/WarpXAMReXInit.cpp: In function ‘amrex::AMReX* warpx::initialization::amrex_init(int&, char**&, bool)’:
/home/runner/work/warpx/warpx/Source/Initialization/WarpXAMReXInit.cpp:185:17: error: ‘MPI_COMM_WORLD’ was not declared in this scope; did you mean ‘amrex::MPI_COMM_WORLD’?
  185 |                 MPI_COMM_WORLD,
      |                 ^~~~~~~~~~~~~~
      |                 amrex::MPI_COMM_WORLD
In file included from /home/runner/work/warpx/warpx/build/_deps/fetchedamrex-src/Src/Base/AMReX.H:9,
                 from /home/runner/work/warpx/warpx/build/_deps/fetchedamrex-src/Src/Base/AMReX_BLassert.H:6,
                 from /home/runner/work/warpx/warpx/build/_deps/fetchedamrex-src/Src/Base/AMReX_ParmParse.H:5,
                 from /home/runner/work/warpx/warpx/Source/Utils/Parser/ParserUtils.H:12,
                 from /home/runner/work/warpx/warpx/Source/Initialization/WarpXAMReXInit.cpp:10:
/home/runner/work/warpx/warpx/build/_deps/fetchedamrex-src/Src/Base/AMReX_ccse-mpi.H:56:22: note: ‘amrex::MPI_COMM_WORLD’ declared here
   56 | static constexpr int MPI_COMM_WORLD    = 0;
      |                      ^~~~~~~~~~~~~~
/home/runner/work/warpx/warpx/Source/Initialization/WarpXAMReXInit.cpp: At global scope:
/home/runner/work/warpx/warpx/Source/Initialization/WarpXAMReXInit.cpp:103:5: error: ‘void {anonymous}::overwrite_amrex_parser_defaults()’ defined but not used [-Werror=unused-function]
  103 |     overwrite_amrex_parser_defaults ()
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
gmake[2]: *** [CMakeFiles/lib_1d.dir/build.make:1762: CMakeFiles/lib_1d.dir/Source/Initialization/WarpXAMReXInit.cpp.o] Error 1
```

--- a/Src/Base/AMReX_ccse-mpi.H
+++ b/Src/Base/AMReX_ccse-mpi.H
@@ -43,6 +43,8 @@ namespace amrex::ParallelDescriptor
 
 #else
 
+namespace amrex {
+
 using MPI_Op       = int;
 using MPI_Comm     = int;
 using MPI_Group    = int;
@@ -59,6 +61,8 @@ static constexpr int MPI_MAX_PROCESSOR_NAME = 64;
 
 static constexpr int MPI_MINLOC = 0;
 static constexpr int MPI_MAXLOC = 0;
+
+}
 
 namespace amrex::ParallelDescriptor
 {

--- a/Tests/MultiBlock/Advection/main.cpp
+++ b/Tests/MultiBlock/Advection/main.cpp
@@ -7,6 +7,8 @@
 
 #include "AMReX_PlotFileUtil.H"
 
+using namespace amrex;
+
 void MyMain();
 
 int main(int argc, char** argv) {
@@ -23,8 +25,6 @@ int main(int argc, char** argv) {
     MPI_Finalize();
 #endif
 }
-
-using namespace amrex;
 
 enum idirs { ix, iy };
 enum num_components { three_components = 3 };

